### PR TITLE
:seedling: cache: moves common HTTP handlers to a shared pkg

### PR DESCRIPTION
**Pull request by p0lyn0mial**
## Summary
It moves common HTTP handlers to a separate package.
Those handlers are used by the kcp-server and the cache-server.
Since we are going to run the cache from the kcp.
Importing will cause a cycle. 
Thus common functionality is moved to a common package.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by sttts** [comment]
/hold

Wrong direction. We should not share the handler chain between kcp and the caching server. Why should we?

File: pkg/server/handler.go [hunk 1]
```diff
@@ -184,19 +78,6 @@ func WithAuditAnnotation(handler http.Handler) http.HandlerFunc {
 	})
 }
 
-// WithClusterAnnotation adds the cluster name into the annotation of an audit
-// event. Needs initialized annotations.
-func WithClusterAnnotation(handler http.Handler) http.HandlerFunc {
```
Comment (sttts) [inline pkg/server/handler.go:189]: this needs a better name. WithAuditEventClusterAnnotation

File: pkg/server/config.go [hunk 2]
```diff
@@ -322,12 +323,12 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
...
-		apiHandler = WithClusterAnnotation(apiHandler)
+		apiHandler = util.WithClusterAnnotation(apiHandler)
 		apiHandler = WithAuditAnnotation(apiHandler) // Must run before any audit annotation is made
-		apiHandler = WithClusterScope(apiHandler)
+		apiHandler = util.WithClusterScope(apiHandler)
 		apiHandler = WithInClusterServiceAccountRequestRewrite(apiHandler)
-		apiHandler = WithAcceptHeader(apiHandler)
-		apiHandler = WithUserAgent(apiHandler)
+		apiHandler = util.WithAcceptHeader(apiHandler)
+		apiHandler = util.WithUserAgent(apiHandler)
```
[the remark this answers is not in the document]
Comment (p0lyn0mial) [reply pkg/server/config.go]: `WithUserAgent` wasn't required also was able to get rid of `WithAcceptHeader`

File: pkg/server/handler.go (hunk 1 above)
Comment (p0lyn0mial) [reply pkg/server/handler.go:189]: updated.

**Comment by p0lyn0mial** [comment]
added `lgtm` label manually since I only fixed the boilterplate

**Comment by stevekuznetsov** [comment]
https://github.com/kcp-dev/kcp/issues/342

**Pull request merged**